src/runtime_src/core/tools/common/ReportAieShim.cpp

```diff
@@ -0,0 +1,338 @@
+/*
+  Copyright (C) 2020 Xilinx, Inc
+ 
+  Licensed under the Apache License, Version 2.0 (the "License"). You may
+  not use this file except in compliance with the License. A copy of the
+  License is located at
+ 
+      http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+  License for the specific language governing permissions and limitations
+  under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "ReportAieShim.h"
+#include "core/common/query_requests.h"
+#include "core/common/device.h"
+#include <boost/optional/optional.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+#define fmt4(x) boost::format("    %-22s: " x "\n")
+#define fmt8(x) boost::format("        %-22s: " x "\n")
+#define fmt12(x) boost::format("            %-22s: " x "\n")
+#define fmt16(x) boost::format("                %-22s: " x "\n")
+
+namespace qr = xrt_core::query;
+
+boost::property_tree::ptree
+populate_aie_shim(const xrt_core::device * device, const std::string& desc)
+{
+  boost::property_tree::ptree pt;
+  std::string aie_data;
+  std::vector<std::string> graph_status;
+  pt.put("description", desc);
+  boost::property_tree::ptree tile_array;
+  boost::property_tree::ptree _pt;

```

Comment:
It is evident that this code was copied from other functionality by how this variable is names.   In this case, the _pt variable was a return parameter.

For this method, a property tree is being returned.  Doing so does cause a copy to occur.   Is there a reason why that the property tree cannot be return via a parameter instead?

Reply:
fixed